lib/mini_profiler/profiler.rb

```diff
@@ -76,6 +76,35 @@ def initialize(app, config = nil)
         @config.storage_instance = @config.storage.new(@config.storage_options)
       end
       @storage = @config.storage_instance
+      if defined?(ActiveSupport)
+        ActiveSupport::Notifications.subscribe "sql.active_record" do |*args|
```

Comment:
sadly this is far from a zero cost thing @nateberkopec , in fact at Discourse we do stuff like this:

https://github.com/discourse/discourse/blob/b5705348b32fff406d8e7e3463def65eaf635206/config/initializers/300-perf.rb#L3-L12

I am ok with adding this stuff to rack-mini-profiler, but we are going to new a setting here and to default it off explaining that this adds cost if you turn it on. 

Ideally we should measure what the actual cost of this extra subscriber is.

Reply:
Interesting!

I wonder if there's a better way to get this stuff, because all we care about is the caller locations when QueryCache is hit. We don't actually have to subscribe to _all_ AR notifications.

Reply:
> I am ok with adding this stuff to rack-mini-profiler, but we are going to new a setting here and to default it off explaining that this adds cost if you turn it on.

I mean this is fine with me too tbh

Reply:
I will add a configuration option and default it to be off.

Reply:
I think ideally it would be in pp=help and work the same way no-backtrace, normal-backtrace and full-backtrace does. Maybe call it no-ar-cache (default), show-ar-cache (option).

That way you could still use it in production and just flip it back off when you're done. You can unsubscribe from ActiveSupport notifications, right?